### DISCUSSION TITLE
New simpler rounding function and new `SHA256` for the output

### DISCUSSION THREAD

**Discussion by gcarreno** [ObjectPascal-Community/1brc-ObjectPascal#62] [Announcements]
Hey Y'All,

After a bit of talking to a prospect entry person, I've come to the realisation that we had a very complicated rounding implementation.

The conclusion of that talk and further code is now the new official rounding code.

I've also altered the `README.md` file to include the code and the new `SHA256` hash of the output, alongside an archived file containing the new baseline output.

Cheers,
Gus

**Comment by synopse** [comment]
I have implemented it for the mormot entry, and it seems to generate the expected output.

As we already discussed, I also prefer this new ceil() method to the previous implementation, which was based on the weird [and buggy](https://stackoverflow.com/questions/9902968/why-does-math-round0-49999999999999994-return-1/9903777#9903777) Java `Math.round()` function.

**Reply by gcarreno** [reply]
Hey Arnaud,

Indeed I agree.
When I looked at the elegance of `Result:= (Ceil(x * 10)) / 10`, a true one liner, it just made absolute sense.
Then, looking at the test results that they provided, it all clicked :smile:

Cheers,
Gus
